tests/test_schema.py

```diff
@@ -246,30 +308,464 @@ class Model(BaseModel):
+        c: Dict[str, Bar]
+
+    flat_models = get_flat_models_from_model(Baz)
+    assert flat_models == set([Foo, Bar, Baz])
+
+
+def test_flat_models_with_submodels_from_sequence():
+    class Foo(BaseModel):
+        a: str
+
+    class Bar(BaseModel):
+        b: Foo
+
+    class Baz(BaseModel):
+        c: Bar
+
+    class Ingredient(BaseModel):
+        name: str
+
+    class Pizza(BaseModel):
+        name: str
+        ingredients: List[Ingredient]
+
+    flat_models = get_flat_models_from_models([Baz, Pizza])
+    assert flat_models == set([Foo, Bar, Baz, Ingredient, Pizza])
+
+
+def test_model_name_maps():
+    class Foo(BaseModel):
+        a: str
+
+    class Bar(BaseModel):
+        b: Foo
+
+    class Baz(BaseModel):
+        c: Bar
+
+    flat_models = get_flat_models_from_models([Baz, ModelA, ModelB, ModelC, ModelD])
+    name_model_map, model_name_map = get_model_name_maps(flat_models)
+    assert name_model_map == {
+        'Foo': Foo,
+        'Bar': Bar,
+        'Baz': Baz,
+        'tests__schema_test_package__modulea__modela__Model': ModelA,
+        'tests__schema_test_package__moduleb__modelb__Model': ModelB,
+        'tests__schema_test_package__moduled__modeld__Model': ModelD,
+    }
+    assert model_name_map == {
+        Foo: 'Foo',
+        Bar: 'Bar',
+        Baz: 'Baz',
+        ModelA: 'tests__schema_test_package__modulea__modela__Model',
+        ModelB: 'tests__schema_test_package__moduleb__modelb__Model',
+        ModelD: 'tests__schema_test_package__moduled__modeld__Model',
+    }
+
+
+def test_schema_overrides():
+    class Foo(BaseModel):
+        a: str
+
+    class Bar(BaseModel):
+        b: Foo = Foo(a='foo')
+
+    class Baz(BaseModel):
+        c: Optional[Bar]
+
+    class Model(BaseModel):
+        d: Baz
+
+    model_schema = Model.schema()
+    assert model_schema == {
+        'title': 'Model',
+        'type': 'object',
+        'definitions': {
+            'Bar': {
+                'title': 'Bar',
+                'type': 'object',
+                'properties': {
+                    'b': {
+                        'title': 'Foo',
+                        'type': 'object',
+                        'properties': {'a': {'title': 'A', 'type': 'string'}},
+                        'required': ['a'],
+                        'default': {'a': 'foo'},
+                    }
+                },
+            },
+            'Baz': {'title': 'Baz', 'type': 'object', 'properties': {'c': {'$ref': '#/definitions/Bar'}}},
+        },
+        'properties': {'d': {'$ref': '#/definitions/Baz'}},
+        'required': ['d'],
+    }
+
+
+def test_schema_from_models():
+    class Foo(BaseModel):
+        a: str
+
+    class Bar(BaseModel):
+        b: Foo
+
+    class Baz(BaseModel):
+        c: Bar
+
+    class Model(BaseModel):
+        d: Baz
+
+    class Ingredient(BaseModel):
+        name: str
+
+    class Pizza(BaseModel):
+        name: str
+        ingredients: List[Ingredient]
```

Comment:
do we really need all these models to test the functionality? Unit tests should be as simple as possible while testing the desired behaviour.

Reply:
I removed one of the submodels. But having at least two models with submodels would verify that the submodels are included not only for the first model in the list. Let me know if you still want me to reduce more.